rules/opentitan_test.bzl

```diff
@@ -452,3 +454,138 @@ def opentitan_functest(
             # For more see https://bazel.build/reference/be/general#test_suite.tags
         ],
     )
+
+def _opentitan_sram_fpga_cw310_test(ctx):
+    test_script = '''#!/usr/bin/env bash
+set -e
+
+COLOR_RED='\\x1b[0;31m'
+COLOR_GREEN='\\x1b[0;32m'
+COLOR_PURPLE='\\x1b[0;35m'
+
+function prefix_lines() {{
+    LABEL="$1"
+    COLOR="$2"
+    COLOR_RESET='\\x1b[m'
+    sed -Eu "s/(.*)/$COLOR[$LABEL]$COLOR_RESET \\1/"
+}}
+
+set -x
+
+./opentitantool load-bitstream --rom-kind=TestRom test_rom.bit
+
+(openocd -f /usr/share/openocd/scripts/interface/ftdi/olimex-arm-usb-tiny-h.cfg \\
+    -c "adapter speed 500; transport select jtag; reset_config trst_and_srst" \\
+    -f {openocd_earlgrey_config} 2>&1 \\
+    | prefix_lines OPENOCD "$COLOR_PURPLE") &
+OPENOCD_PID=$!
+
+(/tools/riscv/bin/riscv32-unknown-elf-gdb --command=gdb_script 2>&1 \\
+    | prefix_lines GDB "$COLOR_GREEN") &
+
+./opentitantool console \\
+    --timeout 15s \\
+    --exit-success='{exit_success_pattern}' \\
```

Comment:
nit: We have a e2e test that checks if the debug access is disabled. I think we might be able to just wait the for the regular shutdown output, i.e. `BFV` & `LCV`, but I'm not sure how the previous commands will behave.

Reply:
Hmm, I didn't quite follow this comment. Is the issue that this `--exit-success=` parameter is not sufficiently generic for other (future) GDB-based tests?

Reply:
Sorry, what I was trying to say is, this whole script might work for the e2e test that I mentioned (debug access disabled in certain lifecycle states) but we need to check how other commands will behave, e.g. return code of openOCD.

Reply:
Oh, gotcha. I'll mark this resolved since the concern seems to already be noted in #14484 ("Check if existing test covers most or all of this testpoint").